base/linalg/matmul.jl

```diff
@@ -44,7 +44,7 @@ end
 vecdot{T<:BlasReal}(x::Union{DenseArray{T},StridedVector{T}}, y::Union{DenseArray{T},StridedVector{T}}) = BLAS.dot(x, y)
 vecdot{T<:BlasComplex}(x::Union{DenseArray{T},StridedVector{T}}, y::Union{DenseArray{T},StridedVector{T}}) = BLAS.dotc(x, y)
 
-function dot{T<:BlasReal, TI<:Integer}(x::Vector{T}, rx::Union{UnitRange{TI},Range{TI}}, y::Vector{T}, ry::Union{UnitRange{TI},Range{TI}})
+function dot{T<:BlasReal, TI<:Integer}(x::StridedVector{T}, rx::Union{UnitRange{TI},Range{TI}}, y::StridedVector{T}, ry::Union{UnitRange{TI},Range{TI}})
```

Comment:
Additionally, isn't the following 
```julia
function dot{T<:BlasReal}(DX::Union{DenseArray{T},StridedVector{T}}, DY::Union{DenseArray{T},StridedVector{T}})
```
rather strange, as `DenseArray <: StridedVector == true`? I mean, wouldn't `StridedVector` suffice?